proxy/Main.cc

```diff
@@ -1901,28 +1884,19 @@ main(int /* argc ATS_UNUSED */, const char **argv)
       if (delay_p && ink_atomic_cas(&delay_listen_for_cache_p, 0, 1)) {
         Debug("http_listen", "Delaying listen, waiting for cache initialization");
       } else {
-        start_HttpProxyServer(); // PORTS_READY_HOOK called from in here
+        eventProcessor.schedule_spawn(&initialize_thread_for_accept, ET_NET);
```

Comment:
What happened to calling `start_HttpProxyServer`? That seems a bit of an oversight.

Reply:
So the problem before is that `initialize_thread_for_net` is called after `start_HttpProxyServer` so the PollDescriptor is null, and since `initialize_thread_for_net` is scheduled at the very start of ET_NET scheduling the `initialize_thread_for_accept` to prepare data then `start_HttpProxyServer` won't make sense since it still might not be ready. What I did here is duplicated the logic in `start_HttpProxyServer` inside `initialize_thread_for_accept`, but just for the part where accept threads is set to 0, so it starts the whole initialization process and all the other stuff when ET_NET starts.